src/client/app/components/visual-group/CreateVisualGroupComponent.tsx

```diff
@@ -0,0 +1,851 @@
+                sortedGroups.push(column);
+                return;
+            }
+
+            const currentSortedColumn: GroupData[] = [];
+
+            if (columnIndex < 1) {
+                for (const meter of meters) {
+                    const parentGroups = mergedGroups.filter(group => group.childMeters.includes(meter.id) && column.includes(group));
+
+                    /*Saving the length before pushing current parent group. 
+                    This ensures that a group with a single meter child is 
+                    placed closer to it's meter child*/
+                    const currentLength = currentSortedColumn.length;
+                    for (const parent of parentGroups) {
+                        if (currentSortedColumn.includes(parent)) continue;
+
+                        /*If any parent group has the meter as their only child, then insert that parent 
+                        before any of the other parents currently in parentGroup. To ensure less edge interceptions*/
+                        if (parent.childMeters.length == 1 && currentLength > 0) {
+                            currentSortedColumn.splice(currentLength, 0, parent);
+                            continue;
+                        }
+
+
+                        currentSortedColumn.push(parent);
+                    }
+                }
+            }
+            else {
+
+                const previousColumn: GroupData[] = columns[columnIndex - 1];
+
+
+                for (const currentGroup of previousColumn) {
+                    const parentGroups = mergedGroups.filter(group => group.childGroups.includes(currentGroup.id) && column.includes(group));
+
+
+                    if (parentGroups.length === 0) continue;
+
+                    /*Saving the length before pushing current parent group. 
+                    This ensures that a group with a single meter child is 
+                    placed closer to it's meter child*/
+                    const currentLength = currentSortedColumn.length;
+                    for (const parent of parentGroups) {
+                        if (currentSortedColumn.includes(parent)) continue;
+
+                        /*If any parent group has the meter as their only child, then insert that parent 
+                        before any of the other parents currently in parentGroup. To ensure less edge interceptions*/
+                        if (parent.childGroups.length == 1 && currentLength > 0) {
+                            currentSortedColumn.splice(currentLength, 0, parent);
+                            continue;
+                        }
+
+
+                        currentSortedColumn.push(parent);
+                    }
+
+                }
+
+
+
+            }
+
+            sortedGroups.push(currentSortedColumn);
+        });
+
+        return sortedGroups;
+    };
+
+    const sortedFinally = sortGroupsByChildren(columns, sortedGroupedMeters);
+
+    /*visuals start here */
+    useEffect(() => {
```

Comment:
I looked and the unit graphic also has it around a useEffect but I wonder about this. Maybe it is needed for d3 but I'm commenting that someone should look into this. If you don't know then a TODO comment would be good.